### PR TITLE
feat: use correct remote name of the current branch

### LINKED ISSUE #2484: Allow to specify the git remote name
### Discussed in https://github.com/goreleaser/goreleaser/discussions/2401

<div type='discussions-op-text'>

<sup>Originally posted by **misitebao** August 14, 2021</sup>
I want to know how to specify the name of the remote github warehouse in the configuration file. The default is origin, but if my warehouse is set to another name, the packaging will fail!</div>

---

right now it defaults to origin on `internal/git/config.go`, maybe we should allow to customize that?

### PR DESCRIPTION
**Pull request by developer-guy**
Signed-off-by: Batuhan Apaydın <batuhan.apaydin@trendyol.com>
Co-authored-with: Erkan Zileli <erkan.zileli@trendyol.com>
Co-authored-with: Furkan Türkal <furkan.turkal@trendyol.com>

to use the correct remote name of the current branch #2486

...

### CODE REVIEW

**Comment by caarlos0** [comment]
this actually gets the first remote it finds, instead of allowing to customize, correct? 

I'm fine with it as well, just would then change title/desc of the pr

And once again, thanks for the hard work on this! 🙏

**Comment by developer-guy** [comment]
> this actually gets the first remote it finds, instead of allowing to customize, correct?

**Exactly, it founds the remote name of the current branch instead of allowing to customize**

> I'm fine with it as well, just would then change title/desc of the pr

**Of course.**

> And once again, thanks for the hard work on this! 🙏

**it's a pleasure.**

File: internal/testlib/git.go [hunk 1]
```diff
@@ -20,9 +20,9 @@ func GitInit(tb testing.TB) {
 }
 
 // GitRemoteAdd adds the given url as remote.
-func GitRemoteAdd(tb testing.TB, url string) {
+func GitRemoteAdd(tb testing.TB, remote, url string) {
```
Comment (caarlos0) [inline internal/testlib/git.go:23]: maybe create another method ` GitRemoteAddWithName`  to avoid having to repeat ` origin`  in a lot of places? wdyt?

Comment (developer-guy) [reply internal/testlib/git.go]: of course, makes sense 🤝

**Review by caarlos0 — APPROVED** [review]
awesome, thxx

**Comment by caarlos0** [comment]
seems like there are some broken tests, can you fix them?

**Comment by developer-guy** [comment]
> seems like there are some broken tests, can you fix them?

thank you @caarlos0, it should be fixed now.

**Comment by caarlos0** [comment]
seems like 2 tests still failing 🤔

**Review by caarlos0 — APPROVED** [review]

**Comment by developer-guy** [comment]
I think we are also ready for merging this PR @caarlos0, PTAL 🤝🥳

**Pull request merged**